### PR TITLE
Refactor + Support for Square and Hexagon cells

### LINKED ISSUE #3: Hex and square cartogram grids

### PR DESCRIPTION
**Pull request by addu390**
Resolves #3

### CODE REVIEW

**Comment by addu390** [comment]
The PR Includes:
- Support for Hexagon and Squares, the shaper.js file has all the changes necessary logic to change when the shape is different.
- Splitting the functions to `events.js` (currently has all the mouse events), `plot.js` (responsible for plotting and scaling the cartogram), `export.js` (To export the cartogram - expected to grow over time), `constants.js` (all the constants in one place), `cartogram.js` (Algorithm to distort the map to form a cartogram).

File: core/plot.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    var topoFeatures = topo_cartogram(topo, topo.objects.tiles.geometries).features
+    export_format(topoFeatures);
+
+    let newHexbin = hexbin()
+        .radius(radius)
+        .x(function (d) { return d.x; })
+        .y(function (d) { return d.y; })
+
+    d3.select('#container').selectAll("*").remove()
```
Comment (ikesau) [inline core/plot.js:49]: What does `selectAll("*")` do? Select every child of `#container`?

File: core/plot.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+function getData(data) {
+    var obj = {}
+    for (var x in data) {
+        obj[data[x].code] = data[x]
+    }
+    return obj
+}
+
+function flatten_features(topoFeatures) {
```
Comment (ikesau) [inline core/plot.js:111]: There are quite a few sneaky snake cases throughout the code. Might want to pick a style to stick with 😁

File: core/plot.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        .style('stroke', '#e0e0e0')
+        .style('stroke-width', strokeWidth)
+        .on("click", mclickBase);
+
+    let features = flatten_features(topoFeatures);
+    for (let i = 0; i < features.length; i++) {
+        for (let j = 0; j < features[i].coordinates.length; j++) {
+            var polygonPoints = features[i].coordinates[j];
+
+            let usPoints = pointGrid.reduce(function (arr, el) {
```
Comment (ikesau) [inline core/plot.js:74]: What is `usPoints` ? The points that are inside the current feature? 

A clearer name would be helpful 🙂

File: core/plot.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+                if (d3.polygonContains(polygonPoints, [el.x, el.y])) arr.push(el);
+                return arr;
+            }, [])
+
+            svg.append('g')
+                .attr('id', 'hexes')
+                .selectAll('.hex')
+                .data(getGridData(cellShape, newHexbin, usPoints))
+                .enter().append('path')
+                .attr('class', 'hex' + features[i].properties.id)
```
Comment (ikesau) [inline core/plot.js:84]: Is this missing a space between `hex` and the `id` ?

File: core/plot.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                .on("mouseout", mout)
+                .call(d3.drag()
+                    .on("start", dragstarted)
+                    .on("drag", dragged)
+                    .on("end", dragended));
+        }
+    }
+}
+
+function getData(data) {
```
Comment (ikesau) [inline core/plot.js:103]: `indexByCode` or something might be a better name

File: core/shaper.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        + "h" + (width - radius)
+        + "a" + radius + "," + radius + " 0 0 1 " + radius + "," + radius
+        + "v" + (height - 2 * radius)
+        + "a" + radius + "," + radius + " 0 0 1 " + -radius + "," + radius
+        + "h" + (radius - width)
+        + "z";
+}
+
+export function getRadius(radius, cellShape) {
+    switch (cellShape) {
```
Comment (ikesau) [inline core/shaper.js:14]: Nice. This pattern throughout these functions is super clear.

**Review by ikesau — COMMENTED** [review]
Hey Adesh! Just a couple of nitpicky comments and questions from a first pass through. Looking good 🔥 

I'd recommend running `prettier` on the JavaScript code (and using an editor like VSCode that has a prettier extension) just to standardize things.

Once that's done I'll give things another go through in more depth 👍

File: core/plot.js (hunk 4 above)
Comment (addu390) [reply core/plot.js:84]: Yes, this is the class name to group a set of hexes together belonging to a polygon/country (example: `hex346`)

File: core/plot.js (hunk 2 above)
Comment (addu390) [reply core/plot.js]: Agreed, there were two other, following camel case everywhere from now on.

File: core/plot.js (hunk 3 above)
Comment (addu390) [reply core/plot.js]: Renamed to `tessellatedPoints` points. Not sure why I named it `usPoints`

**Review by ikesau — APPROVED** [review]
:shipit:

**Pull request merged**